lib/hanami/repository.rb

```diff
@@ -153,10 +153,15 @@ def self.container
     # @api private
     def self.define_relation # rubocop:disable Metrics/MethodLength
       a = @associations
+      s = @schema
 
       configuration.relation(relation) do
-        schema(infer: true) do
-          associations(&a) unless a.nil?
+        if s.nil?
+          schema(infer: true) do
+            associations(&a) unless a.nil?
```

Comment:
@mereghost calling `schema(infer: true)` is required, only the block is optional